app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js

```diff
@@ -697,6 +697,7 @@ define([
                 $label = $parent.find('.' + $widget.options.classes.attributeSelectedOptionLabelClass),
                 attributeId = $parent.attr('attribute-id'),
                 $input = $parent.find('.' + $widget.options.classes.attributeInput);
+                checkAdditionalData = JSON.parse(this.options.jsonSwatchConfig[attributeId].additional_data);
```

Comment:
It looks like an unnecessary variable. Even if it's used for readability it'd be better to move the declaration closer to the usage. Also, please use `const` statement to declare this variable

Reply:
Hello @sivaschenko 
we can't use `const` because checkAdditionalData each time load only current attribute data only like if click on size swatch load only size data or click on color swatch load only color swatch data

Reply:
@ravi-chandra3197 I see only one assignment to this variable in the scope of the function, so the variable is immutable, `const` is fine from my point of view (even if was immutable the variable availability scope should be limited by `let` statement)
Please move the variable declaration to line 732 to be as close as possible to the usage of this variable